bin/insertMissingAlerts.ts

```diff
@@ -4,14 +4,31 @@ import { Alerts, User } from '../src/entity';
 
 (async () => {
   const con = await createOrGetConnection();
-  const usersQuery = con
+  const stream = await con
```

Comment:
This will not work as expected since you have an async function inside a stream. It will introduce race conditions. please look at Ante's cron job for the digest to see how to implement it correctly. It's not a big change but crucial one

Reply:
I see. Cool. In this case, I assume we will batch the insertion as well, right?

Reply:
This would mean, we will not need the stream anymore, since we can fetch all the ids in a single query unlike reading from a file which is handled in the stream.

Reply:
We still stream but we use a library to help us with the async nature of our workload. Take a look on that script and you will see. I'm talking specifically about the cron job for the digest

Reply:
This one: https://github.com/dailydotdev/daily-api/blob/main/src/cron/personalizedDigest.ts

Reply:
Ah, but note that this is not a cron job and only a script to run locally while connecting to prod. A one time thing kind of script.

Reply:
I understand but it will still suffer from the same race condition hence, some users eventually will not have a alerts set properly. The problem is that the script will finish before the async functions are done with their job. Hence it will introduce inconsistency

Reply:
Got it. Will refactor the process to that.